x11rb-async/src/rust_connection/stream.rs

```diff
@@ -60,7 +60,12 @@ impl<S> StreamAdaptor<S> {
     }
 
     /// Get a mutable reference to the inner stream.
-    pub fn get_mut(&mut self) -> &mut S {
+    ///
+    /// # Safety
+    ///
+    /// This function inherits its unsafety from [`async_io::Async::get_mut`]. This means that the
+    /// underlying I/O source must not be dropped using this function.
+    pub unsafe fn get_mut(&mut self) -> &mut S {
```

Comment:
Do we use this function for anything? If not I'd rather just remove it

Reply:
If I remove the `pub`, this causes warnings from `warn(dead_code)`. So, no, we don't use these anywhere (everything still builds).

The `StreamAdaptor` type also doesn't seem to be publically accessible. It is only used in `RustConnection::connect()` (through `StreamAdaptor::new()`). The result is then given to `RustConnection::connect_to_stream_with_auth_info()` which ends up passing this to `SharedState::new()`, which in turn doesn't allow to access the inner `stream`.

Thus, there is no way to get the `Stream` from a `RustConnection` and thus this `pub` is actually pretty much unreachable (unless one uses the `StreamAdaptor` externally for "other things", but then... yeah. I can live with that API break).

New commit pushed that removes everything but the API that is actually required from `StreamAdaptor`.